onnxruntime/core/providers/shared/utils/utils.cc

```diff
@@ -53,7 +55,11 @@ bool GetClipMinMax(const InitializedTensorSet& initializers, const Node& node,
         return false;
       }
       Initializer unpacked_tensor(*initializers.at(min_name));
-      min = unpacked_tensor.DataAsSpan<float>()[0];
+      if (input_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
```

Comment:
I feel switch...case is a little bit better in case later one someone modified line 35 but didn't notice this line of code.

Reply:
Switch is nice.

I'm surprised the utility function isn't just generic, like the DML EP's general form:

[https://github.com/microsoft/onnxruntime/blob/cf974f0905a78280c1a47b519e7d6c0b0a6ca069/onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlCommon.h#L31](https://github.com/microsoft/onnxruntime/blob/cf974f0905a78280c1a47b519e7d6c0b0a6ca069/onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlCommon.h#L31)

Used here:
[https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp#L454](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp#L454)

Reply:
kk, switch it is.

Reply:
fixed.